addons/web/static/tests/core/l10n/translation.test.js

```diff
@@ -79,6 +81,147 @@ test("can translate a text node", async () => {
+    const def = new Deferred();
+    onRpc("/web/webclient/translations/*", async (request) => {
+        await def;
+        expect.step(`hash: ${new URL(request.url).searchParams.get("hash")}`);
+    });
+    TestComponent.template = xml`<div id="main">Hello</div>`;
+    defineParams({
+        translations: frenchTerms,
+    });
+    await mountWithCleanup(TestComponent);
+    expect("#main").toHaveText("Bonjour"); //Don't wait the end of the fetch to render
+    def.resolve();
+    await animationFrame();
+    expect.verifySteps(["hash: 5d62e6f31a2e19f1f128bd7f56d11088a746e001"]); //Fetch with the hash of the translation in cache
+});
+
+test("[cache] update the cache if hash are different", async () => {
+    patchWithCleanup(IndexedDB.prototype, {
+        read() {
+            return {
+                lang: "en",
+                lang_parameters: {
+                    date_format: "%m/%d/%Y",
+                    decimal_point: ".",
+                    direction: "ltr",
+                    grouping: "[3,0]",
+                    time_format: "%H:%M:%S",
+                    short_time_format: "%H:%M",
+                    thousands_sep: ",",
+                    week_start: 7,
+                },
+                modules: { web: { messages: [{ id: "Hello", string: "Different Bonjour" }] } },
+                multi_lang: false,
+                hash: "5d62e6f31a2e",
+            };
+        },
+        write(table, key, value) {
+            expect.step(`table: ${table}`);
+            expect.step(`key: ${key}`);
+            expect.step(`value: ${JSON.stringify(value)}`);
+        },
+    });
+    const def = new Deferred();
+    onRpc("/web/webclient/translations/*", async (request) => {
+        await def;
+        expect.step(`hash: ${new URL(request.url).searchParams.get("hash")}`);
+    });
+    TestComponent.template = xml`<div id="main">Hello</div>`;
+    defineParams({
+        translations: frenchTerms,
+    });
+    await mountWithCleanup(TestComponent);
+    expect("#main").toHaveText("Different Bonjour"); //Value came from the cache!
+    def.resolve();
+    await animationFrame();
+    const expectedValue = {
+        lang: "en",
+        lang_parameters: {
+            date_format: "%m/%d/%Y",
+            decimal_point: ".",
+            direction: "ltr",
+            grouping: "[3,0]",
+            time_format: "%H:%M:%S",
+            short_time_format: "%H:%M",
+            thousands_sep: ",",
+            week_start: 7,
+        },
+        modules: { web: { messages: [{ id: "Hello", string: "Bonjour" }] } }, // value was updated in the cache
+        multi_lang: false,
+        hash: "5d62e6f31a2e19f1f128bd7f56d11088a746e001", // hash was updated in the cache
+    };
+    expect.verifySteps([
+        "hash: 5d62e6f31a2e", //Fetch with the hash of the translation in cache
+        "table: /web/webclient/translations",
+        'key: {"lang":"en"}',
+        `value: ${JSON.stringify(expectedValue)}`,
+    ]);
```

Comment:
we could assert that the value has been updated correctly in the service as well (not only in indexeddb), for instance by re-rendering the component